rust/noosphere-p2p/src/dht/transport.rs

```diff
@@ -0,0 +1,35 @@
+use libp2p::{
+    core::muxing::StreamMuxerBox, core::transport::Boxed, core::upgrade, dns, mplex, noise, tcp,
+    websocket, yamux, PeerId, Transport,
+};
+use std::{io, result::Result};
+
+/// Creates the Transport mechanism that describes how peers communicate.
+/// Currently, mostly an inlined form of `libp2p::tokio_development_transport`.
+pub(crate) fn build_transport(
+    keypair: &libp2p::identity::Keypair,
+) -> Result<Boxed<(PeerId, StreamMuxerBox)>, io::Error> {
+    let transport = {
+        let dns_tcp = dns::TokioDnsConfig::system(tcp::TokioTcpTransport::new(
+            tcp::GenTcpConfig::new().nodelay(true),
+        ))?;
+        let ws_dns_tcp = websocket::WsConfig::new(dns::TokioDnsConfig::system(
```

Comment:
Just curious: does the WebSocket transport help for cases beyond peering with browser nodes?

Reply:
Probably can be removed at least for now, a peer can offer multiple addresses at once, and unclear to me yet how the web version works in context